it/uk/gov/hmrc/minorentityidentificationfrontend/controllers/uaControllers/CheckYourAnswersControllerISpec.scala

```diff
@@ -168,19 +174,129 @@ class CheckYourAnswersControllerISpec extends AuditEnabledSpecHelper
 
       result must have {
         httpStatus(SEE_OTHER)
         redirectUri(expectedValue = s"$testContinueUrl?journeyId=$testJourneyId")
       }
+
+      verifyStoreIdentifiersMatch(testJourneyId, expBody = JsString(SuccessfulMatchKey))
+      verifyAudit()
+    }
+
+    "redirect to the cannot confirm business error page when the identifiers are not matched" in {
+
+      enable(EnableFullUAJourney)
+
+      await(insertJourneyConfig(
+        journeyId = testJourneyId,
+        internalId = testInternalId,
+        testUnincorporatedAssociationJourneyConfig(businessVerificationCheck = true)
+      ))
+
+      stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+      stubRetrieveUtr(testJourneyId)(OK, testCtutrJson)
+      stubRetrievePostcode(testJourneyId)(OK, testOfficePostcode)
+      stubRetrieveCHRN(testJourneyId)(NOT_FOUND)
+
+      stubValidateUnincorporatedAssociationDetails(testCtutr, testOfficePostcode)(OK, Json.obj("matched" -> false))
+
+      stubStoreIdentifiersMatch(testJourneyId, DetailsMismatchKey)(OK)
+
+      stubAudit()
+
+      val result = post(s"/identify-your-unincorporated-association/$testJourneyId/check-your-answers-business")()
+
+      result must have {
+        httpStatus(SEE_OTHER)
+        redirectUri(expectedValue = errorControllers.routes.CannotConfirmBusinessController.show(testJourneyId).url)
+      }
+
+      verifyStoreIdentifiersMatch(testJourneyId, expBody = JsString(DetailsMismatchKey))
+      verifyAudit()
+    }
+
+    "redirect to the cannot confirm business error page when the unincorporated association's details cannot be found" in {
+
+      enable(EnableFullUAJourney)
+
+      await(insertJourneyConfig(
+        journeyId = testJourneyId,
+        internalId = testInternalId,
+        testUnincorporatedAssociationJourneyConfig(businessVerificationCheck = true)
+      ))
+
+      stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+      stubRetrieveUtr(testJourneyId)(OK, testCtutrJson)
+      stubRetrievePostcode(testJourneyId)(OK, testOfficePostcode)
+      stubRetrieveCHRN(testJourneyId)(NOT_FOUND)
+
+      stubValidateUnincorporatedAssociationDetails(testCtutr, testOfficePostcode)(BAD_REQUEST, Json.obj("code" -> "NOT_FOUND",
+        "reason" -> "The back end has indicated that CT UTR cannot be returned"
+      ))
+
+      stubStoreIdentifiersMatch(testJourneyId, DetailsNotFoundKey)(OK)
+
+      stubAudit()
+
+      val result = post(s"/identify-your-unincorporated-association/$testJourneyId/check-your-answers-business")()
+
+      result must have {
+        httpStatus(SEE_OTHER)
+        redirectUri(expectedValue = errorControllers.routes.CannotConfirmBusinessController.show(testJourneyId).url)
+      }
+
+      verifyStoreIdentifiersMatch(testJourneyId, expBody = JsString(DetailsNotFoundKey))
+      verifyAudit()
+    }
+
+    "continue the registration journey" when {
```

Comment:
Maybe change to "redirect to continue url" when. Follows the rest of your pattern

Reply:
Done